test/test_apps/common/test_app_base.h

```diff
@@ -0,0 +1,1094 @@
+/*
+** Copyright (c) 2018-2024 LunarG, Inc.
+**
+** Permission is hereby granted, free of charge, to any person obtaining a
+** copy of this software and associated documentation files (the "Software"),
+** to deal in the Software without restriction, including without limitation
+** the rights to use, copy, modify, merge, publish, distribute, sublicense,
+** and/or sell copies of the Software, and to permit persons to whom the
+** Software is furnished to do so, subject to the following conditions:
+**
+** The above copyright notice and this permission notice shall be included in
+** all copies or substantial portions of the Software.
+**
+** THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+** IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+** FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+** AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+** LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+** FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+** DEALINGS IN THE SOFTWARE.
+*/
+
+#ifndef GFXRECON_TEST_APP_BASE_H
+#define GFXRECON_TEST_APP_BASE_H
+
+#include <cassert>
+#include <cstdio>
+#include <cstring>
+
+#include <optional>
+#include <exception>
+#include <vector>
+#include <string>
+
+#include <vulkan/vulkan_core.h>
+
+#include <SDL3/SDL.h>
```

Comment:
Do we need SDL3 for all test apps?

Reply:
There could be test apps that don't require it, but since this is supposed to be a quick foundation for all test apps, there is going to be functionality that is included in the base that isn't needed for all of them.  For window and surface creation, SDL is quick and easy, so TestAppBase uses it to bootstrap.